pallets/api/src/fungibles/tests.rs

```diff
@@ -62,32 +108,69 @@ fn approve_works() {
 		let owner = ALICE;
 		let spender = BOB;
 
-		create_token_and_mint_to(ALICE, token, ALICE, value);
-		assert_eq!(0, Assets::allowance(token, &ALICE, &BOB));
-		assert_ok!(Fungibles::approve(signed(ALICE), token, BOB, value));
-		assert_eq!(Assets::allowance(token, &ALICE, &BOB), value);
+		// Approves a value to spend.
+		//
+		// Check error works for `Assets::approve_transfer()` in `Greater` match arm.
+		assert_noop!(
+			Fungibles::approve(signed(owner), token, spender, value),
+			AssetsError::Unknown.with_weight(WeightInfo::approve(1, 0))
+		);
+		pallet_assets_create_and_mint_to(owner, token, owner, value);
+		for origin in vec![root(), none()] {
```

Comment:
Would move these as the first checks in each test, as they are the first checks in the dispatchables.